File: frontend/src/types/intelligence.ts
export interface EntityRef {
  id: string;
  name: string;
  type: "ORG" | "LOC" | "PER" | "MISC";
  frequency: number;
}

export interface ClusterBase {
  id: string;
  title?: string;
  summary?: string;
  message_count: number;
  sentiment_score?: number;
  urgency_score?: number;
  updated_at: string;
  first_message_at?: string;
  primary_source_channel_id?: string;
  velocity?: number;
  emergence_score?: number;
  structured_summary?: {
    headline: string;
    bullets: string[];
    context: string;
  };
}

export interface ClusterList extends ClusterBase {}

export interface ClusterDetail extends ClusterBase {
  messages_preview: Array<{
    id: string;
    text: string;
  }>;
  entities: EntityRef[];
  timeline: Array<{
    time: string;
    count: number;
  }>;
}

export interface IntelligenceDashboard {
  hot_topics: ClusterList[];
  top_entities: Record<string, EntityRef[]>;
  global_tension: number;
}

File: backend/app/schemas/intelligence.py
from pydantic import BaseModel
from typing import List, Optional, Dict
from datetime import datetime
from uuid import UUID

class EntityRef(BaseModel):
    id: UUID
    name: str
    type: str
    frequency: int

class ClusterBase(BaseModel):
    id: UUID
    title: Optional[str]
    summary: Optional[str]
    message_count: int
    sentiment_score: Optional[float]
    urgency_score: Optional[int]
    updated_at: datetime
    first_message_at: Optional[datetime]
    primary_source_channel_id: Optional[UUID]

class ClusterList(ClusterBase):
    pass

class ClusterDetail(ClusterBase):
    messages_preview: List[dict]  # Simplified message objects
    entities: List[EntityRef]
    timeline: List[dict] # {time, count}

class IntelligenceDashboard(BaseModel):
    hot_topics: List[ClusterList]
    top_entities: Dict[str, List[EntityRef]]
    global_tension: int
